src/gui/C4LoaderScreen.cpp

```diff
@@ -113,29 +104,25 @@ void C4LoaderScreen::SetBlackScreen(bool fIsBlack)
 	// will be updated when drawn next time
 }
 
-int C4LoaderScreen::SeekLoaderScreens(C4Group &rFromGrp, const char *szWildcard, int iLoaderCount, char *szDstName, C4Group **ppDestGrp)
+void C4LoaderScreen::SeekLoaderScreens(C4Group &rFromGrp, const std::string wildcard)
 {
-	bool fFound;
-	int iLocalLoaders=0;
-	char Filename[_MAX_PATH+1];
-	for (fFound=rFromGrp.FindEntry(szWildcard, Filename); fFound; fFound=rFromGrp.FindNextEntry(szWildcard, Filename))
+	// seek for png, jpg, jpeg, bmp
+	char filename[_MAX_PATH + 1];
+	for (bool found = rFromGrp.FindEntry(wildcard.c_str(), filename); found; found = rFromGrp.FindNextEntry(wildcard.c_str(), filename))
 	{
-		// loader found; choose it, if Daniel wants it that way
-		++iLocalLoaders;
-		if (!UnsyncedRandom(++iLoaderCount))
-		{
-			// copy group and path
-			*ppDestGrp=&rFromGrp;
-			SCopy(Filename, szDstName, _MAX_PATH);
+		// potential candidate - check file extension
+		std::string extension{ GetExtension(filename) };
+		if (extension == "png" || extension == "jpg" || extension == "jpeg" || extension == "bmp") {
+			auto loader = std::pair<C4Group*, std::string>{ &rFromGrp, std::string(filename) };
+			loaders.emplace(loader);
```

Comment:
Can't you do this directly as `loaders.emplace(&rFromGrp, std::string(filename));`?